lib/logging/monitor.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Logging
+  class Monitor
+    def initialize(service)
+      @service = service
+    end
+
+    ##
+    # log GET request
+    #
+    # @param error_level [String]
+    # @param message [String]
+    # @param metric [String]
+    # @param additional_context [Hash]
+    #
+    def track_request(error_level, message, metric, additional_context, call_location: nil)
+      function, file, line = parse_caller(call_location)
+
+      StatsD.increment(metric, tags: additional_context[:tags])
+
+      if %w[debug info warn error fatal unknown].include?(error_level)
+        Rails.logger.public_send(error_level, message.to_s,
+                                 {
+                                   statsd: metric,
+                                   service:,
+                                   user_account_uuid: additional_context[:user_account_uuid],
+                                   function:,
+                                   file:,
+                                   line:,
+                                   additional_context:
+                                 })
+      else
+        Rails.logger.error("Invalid log error_level: #{error_level}")
+      end
+    end
+
+    private
+
+    attr_reader :service
+
+    def parse_caller(call_location)
+      call_location ||= caller_locations.second
```

Comment:
Is it possible for caller_locations.second to be null? If so, should protect the next line from null call_location values

Reply:
@TaiWilkin It can only be `nil` if this method is called directly from the console. If called from another method which it will always be it won't ever be nil